edx_django_utils/cache/utils.py

```diff
@@ -14,6 +16,27 @@
 _CACHE_MISS = object()
 
 
+def get_cache_key(**kwargs):
+    """
+    Get MD5 encoded cache key for given arguments.
+    Note: This function will not work correctly if any of the arguments are dictionaries
+    Here is the format of key before MD5 encryption.
+        key1:value1__key2:value2 ...
+    Example:
+        >>> get_cache_key(site_domain="example.com", resource="catalogs")
+        # Here is key format for above call
+        # "site_domain:example.com__resource:catalogs"
+        a54349175618ff1659dee0978e3149ca
+    Arguments:
+        **kwargs: Key word arguments that need to be present in cache key.
+    Returns:
+         An MD5 encoded key uniquely identified by the key word arguments.
+    """
+    key = '__'.join([u'{}:{}'.format(item, value) for item, value in sorted(six.iteritems(kwargs))])
```

Comment:
I found the following code from edx-platform: https://github.com/edx/edx-platform/blob/00f5beb66f50bddfbf5a19d26423482056893475/openedx/core/lib/cache_utils.py#L93-L96

The main thing I'd take from this is to use `force_str` (was force_text) in code above, for the values.  It is up to you if you want to take any of the other code.  Optional to add a test with a byte string as one of the argument values.  See https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.encoding.force_str